File: app/src/main/java/com/jrvermeer/psalter/Core/Models/SqLiteQuery.java
package com.jrvermeer.psalter.Core.Models;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by Jonathan on 9/4/2017.
 */

public class SqLiteQuery {
    public SqLiteQuery()
    {
        queryParameters = new ArrayList<String>();
    }

    public String getQueryText() {
        return queryText;
    }
    public void setQueryText(String queryText) {
        this.queryText = queryText;
    }
    private String queryText;

    public void addParameter(String value){
        queryParameters.add(value);
    }

    public String[] getParameters(){
        return queryParameters.toArray(new String[0]);
    }

    private List<String> queryParameters;
}

File: app/src/main/java/com/jrvermeer/psalter/Core/Models/SqLiteQuery.kt
package com.jrvermeer.psalter.Core.Models

import java.util.ArrayList

/**
 * Created by Jonathan on 9/4/2017.
 */

class SqLiteQuery {
    var queryText: String? = null

    val parameters: Array<String>
        get() = queryParameters.toTypedArray()

    private val queryParameters: MutableList<String>

    init {
        queryParameters = ArrayList()
    }

    fun addParameter(value: String) {
        queryParameters.add(value)
    }
}
